emission/tests/funcTests/TestFunctionTiming.py

```diff
@@ -0,0 +1,435 @@
+        function_name = func.__name__
+        logging.info(f"Starting timing for function: {function_name}")
+
+        try:
+            with ec_timer.Timer() as timer:
+                result = func()  # Execute the test function
+                
+            elapsed_ms = (timer.elapsed * 1000)  # Convert to milliseconds
+
+            # Store the execution time
+            sdq.store_dashboard_time(
+                code_fragment_name=function_name,
+                timer=timer
+            )
+            logging.info(f"Function '{function_name}' executed successfully in {elapsed_ms:.2f} ms.")
+
+            # Verification: Query for the specific function's timing entry
+            stored_documents_chain = self.timeseries_db.find_entries(
+                ["stats/dashboard_time"],
+                extra_query_list=[{"data.name": function_name}]
+            )
+
+            # Convert the chain to a list to make it subscriptable and to allow multiple accesses
+            stored_documents = list(stored_documents_chain)
+
+            # Assert that exactly one document was retrieved for the function
+            self.assertEqual(
+                len(stored_documents),
+                1,
+                f"Expected exactly 1 timing entry for '{function_name}', found {len(stored_documents)}."
+            )
+            stored_document = stored_documents[0]
+            # Iterate over each document and inspect its contents
+            try:
+                stored_ts = stored_document['data']['ts']
+                stored_reading = stored_document['data']['reading']
+                stored_name = stored_document['data']['name']
+                logging.info(
+                    f"Stored Document for '{function_name}': ts={stored_ts}, reading={stored_reading} ms, name={stored_name}"
+                )
+            except KeyError as e:
+                self.fail(
+                    f"Missing key {e} in stored document for '{function_name}'."
+                )
+            
+            # Log the expected and actual readings
+            logging.info(
+                f"Function '{function_name}': Expected elapsed time = {elapsed_ms:.2f} ms, "
+                f"Stored reading = {stored_reading} ms."
+            )
+
+            # Assert that the stored_reading matches elapsed_ms exactly
+            self.assertEqual(
+                stored_reading,
+                elapsed_ms,
+                msg=(
+                    f"Stored reading {stored_reading} ms does not exactly match "
+                    f"elapsed time {elapsed_ms:.2f} ms for '{function_name}'."
+                )
+            )
+
+            # Assert that the stored_name matches function_name exactly
+            self.assertEqual(
+                stored_name,
+                function_name,
+                msg=(
+                    f"Stored name '{stored_name}' does not match function "
+                    f"name '{function_name}'."
+                )
+            )
```

Comment:
This is all repeated in `test_code_block_instrumentation`. Let's extract it to a separate function called in both places